fft/src/KokkosFFT_ROCM_plans.hpp

```diff
@@ -129,8 +127,13 @@ auto create_plan(const ExecutionSpace& exec_space,
       convert_int_type_and_reverse<int, std::size_t>(fft_extents);
 
   // Create the description
-  rocfft_plan_description description;
-  rocfft_status status = rocfft_plan_description_create(&description);
+  std::unique_ptr<rocfft_plan_description,
+                  std::function<void(rocfft_plan_description*)>> const
+      description(new rocfft_plan_description,
+                  [](rocfft_plan_description* desc) {
+                    rocfft_plan_description_destroy(*desc);
+                  });
+  rocfft_status status = rocfft_plan_description_create(&(*description));
```

Comment:
I think we can improve the situation by also wrapping `rocfft_plan_description`.

In case `rocfft_plan_description_create` fails, the next `KOKKOSFFT_THROW_IF` will throw, calling the destructor of `description` thus calling `rocfft_plan_description_destroy` even though the `rocfft_plan_description` object has not been initialised.

Reply:
I have introduced a wrapper accordingly